app/src/main/java/com/example/maintenancespace/ui/cars/CarFragment.java

```diff
@@ -26,12 +29,32 @@ public class CarFragment extends Fragment {
 
     private FragmentCarBinding binding;
 
+    public void addCar(CarModel car) {
+        final FragmentManager fragmentManager = getChildFragmentManager();
+        Fragment fragment = fragmentManager.findFragmentById(R.id.car_layout);
+        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
+        CarListItemFragment existingFragment = (CarListItemFragment) fragmentManager.findFragmentByTag(car.getId());
+        if (existingFragment == null) {
+            CarListItemFragment newEventListItem = CarListItemFragment.newInstance(car);
+            fragmentTransaction.add(R.id.car_layout, newEventListItem, car.getId());
+        }
+        fragmentTransaction.detach(fragment);
+        fragmentTransaction.attach(fragment);
+        fragmentTransaction.commitNowAllowingStateLoss();
+    }
+
```

Comment:
This isn't ideal but couldn't find another way to handle it.  It seems to work without issue for what we are doing but may cause issues with other use cases since `commitNowAllowingStateLoss` is being used.